### PR TITLE
Convert nano avax to avax

### PR DESCRIPTION
**Pull request by matthewbauer**
Puts avax in decimal format so 12345 is now 0.000012345.

### CODE REVIEW

File: src/to_string.c [hunk 1]
```diff
@@ -101,7 +93,47 @@ size_t number_to_string(char *const dest, uint64_t number) {
...
+    memcpy(dest + off, start, length);
+    off += length;
+    dest[off] = '\0';
+    return off;
+}
+
+void nano_avax_to_string_indirect64(char *const dest, size_t const buff_size, uint64_t const *const number) {
+    check_null(dest);
+    check_null(number);
+    if (buff_size < MAX_INT_DIGITS + 1)
```
Comment (lpsmith) [inline src/to_string.c:133]: I think this needs to be `MAX_INT_DIGITS + 2`...  one for the null, one for a decimal point.   Otherwise we risk undefined behavior.  (I.e. this appears to be a "type 2" function when it should be a "type 1" function,  in the parlance of [John Regehr](https://blog.regehr.org/archives/213))

Also wondering if this routine should succeed or fail if the buffer isn't big enough for the largest possible int... but that's mostly idle speculation,  as I think this is fine.

File: src/to_string.c [hunk 2]
```diff
@@ -101,7 +93,47 @@ size_t number_to_string(char *const dest, uint64_t number) {
     return length;
 }
 
-#define DECIMAL_DIGITS 8
+#define DECIMAL_DIGITS 9
+#define NANO_AVAX_SCALE 1000000000
+
+// Display avax in human readable form
+size_t nano_avax_to_string(char *const dest, uint64_t nano_avax) {
```
Comment (lpsmith) [inline src/to_string.c:100]: I'm wondering if this function should be marked `static` or not.

File: tests/basic-tests.js [hunk 3]
```diff
@@ -108,8 +108,8 @@ describe("Basic Tests", () => {
       const pathSuffixes = ["0/0", "0/1", "100/100"];
       const ui = await flowMultiPrompt(this.speculos, [
         [{header:"Sign",body:"Transaction"}],
-        [{header:"Transfer",body:"12345 to denali12yp9cc0melq83a5nxnurf0nd6fk4t224dtg0lx"}],
-        [{header:"Fee",body:"123444444"}],
+        [{header:"Transfer",body:"0.000012345 to denali12yp9cc0melq83a5nxnurf0nd6fk4t224dtg0lx"}],
+        [{header:"Fee",body:"0.123444444"}],
```
Comment (lpsmith) [inline tests/basic-tests.js:112]: I'm thinking that a couple more tests should be added to better test the various edge cases,  e.g.:

1.  One that requires the maximum number of digits
2.  One that doesn't print a decimal point
3.  Maybe one for each of the different number of digits shown after the decimal point?
4.  Others?

File: src/to_string.c (hunk 2 above)
Comment (matthewbauer) [reply src/to_string.c:100]: This is referenced in apdu_sign.c

File: src/to_string.c (hunk 1 above)
Comment (matthewbauer) [reply src/to_string.c:133]: MAX_INT_DIGITS is 20, so we can cover up to 10^11 avax (with 10^-9 decimal). So yeah we can't cover anything amounts greater than that.

Comment (lpsmith) [reply src/to_string.c:133]: Right,  a 64 bit integer can be 20 digits.   Then we have one null character at the end,  and one decimal point.   So the destination buffer needs to be be at least 22 bytes long.
